### PR TITLE
ebs br: serial execution of volume backup gc

### LINKED ISSUE #5451: ebs br: serialize deletion of volumebackups and snapshots in backup
## Feature Request

**Is your feature request related to a problem? Please describe:**

For large cluster with larger number of volume backups, concurrent deletion could fail due to AWS API throttling.

**Describe the feature you'd like:**

Make volumebackup and snapshot deletion execution in serial.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by BornChanger**
### What problem does this PR solve?

Closes #5451 

### What is changed and how does it work?

This PR make the EBS snapshot volume backup GC execute serially. In the single volumebackup, delete snapshots in batches.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/fedvolumebackup/backupschedule/backup_schedule_manager.go [hunk 1]
```diff
@@ -258,18 +258,25 @@ func (bm *backupScheduleManager) backupGCByMaxReservedTime(vbs *v1alpha1.VolumeB
...
+			if err = bm.deps.FedVolumeBackupControl.DeleteVolumeBackup(backup); err != nil {
+				klog.Errorf("backup schedule %s/%s gc backup %s failed, err %v", ns, bsName, backup.GetName(), err)
+				return
+			}
+			klog.Infof("backup schedule %s/%s gc backup %s success", ns, bsName, backup.GetName())
 
-	if len(expiredBackups) == len(backupsList) && len(expiredBackups) > 0 {
-		// All backups have been deleted, so the last backup information in the backupSchedule should be reset
-		bm.resetLastBackup(vbs)
+			if len(expiredBackups) == 1 {
```
Comment (WangLe1321) [inline pkg/fedvolumebackup/backupschedule/backup_schedule_manager.go:275]: This condition is not equal to the old one. You should ensure no alive backup before call `bm.resetLastBackup()`

**Review by WangLe1321 — APPROVED** [review]

File: pkg/backup/util/aws_ebs.go [hunk 2]
```diff
@@ -147,10 +148,11 @@ func (e *EC2Session) DeleteSnapshots(snapIDMap map[string]string) error {
 		lock       sync.Mutex
 	)
 
-	eg := new(errgroup.Group)
+	eg, _ := errgroup.WithContext(context.Background())
+	workerPool := NewWorkerPool(e.concurrency, "delete snapshots")
 	for volID := range snapIDMap {
 		snapID := snapIDMap[volID]
-		eg.Go(func() error {
+		workerPool.ApplyOnErrorGroup(eg, func() error {
```
Comment (nkg-) [inline pkg/backup/util/aws_ebs.go:155]: What is the batch size. How many snapshots will be deleted concurrently.

Comment (nkg-) [reply pkg/backup/util/aws_ebs.go:155]: Its 3 - https://github.com/pingcap/tidb-operator/blob/87b28741cf88bec658ff80745a21d13e0150409d/cmd/backup-manager/app/clean/clean.go#L46

**Pull request merged**